lib/PuppeteerSharp/FrameManager.cs

```diff
@@ -67,25 +67,21 @@ public async Task<Response> NavigateFrameAsync(Frame frame, string url, Navigati
             var watcher = new NavigatorWatcher(_client, this, frame, _networkManager, timeout, options);
 
             var navigateTask = NavigateAsync(_client, url, referrer, frame.Id);
-            await Task.WhenAny(
-                watcher.TimeoutOrTerminationTask,
-                navigateTask).ConfigureAwait(false);
+            await Task.Run(() => navigateTask, watcher.TimedCancellationToken).ConfigureAwait(false);
 
-            AggregateException exception = null;
+            Exception exception = null;
             if (navigateTask.IsFaulted)
             {
                 exception = navigateTask.Exception;
             }
             else
             {
-                await Task.WhenAny(
-                    watcher.TimeoutOrTerminationTask,
-                    _ensureNewDocumentNavigation ? watcher.NewDocumentNavigationTask : watcher.SameDocumentNavigationTask
-                ).ConfigureAwait(false);
 
-                if (watcher.TimeoutOrTerminationTask.IsCompleted && watcher.TimeoutOrTerminationTask.Result.IsFaulted)
+                await Task.Run(() => _ensureNewDocumentNavigation ? watcher.NewDocumentNavigationTask : watcher.SameDocumentNavigationTask, watcher.TimedCancellationToken).ConfigureAwait(false);
+
+                if (watcher.TimedCancellationToken.IsCancellationRequested)
                 {
-                    exception = watcher.TimeoutOrTerminationTask.Result.Exception;
+                    exception = new TaskCanceledException("Timed cancellation token timed out.");
```

Comment:
We can throw this exception https://github.com/kblok/puppeteer-sharp/blob/a890badffa63bb8dca833e8b95005e06ed54d7ba/lib/PuppeteerSharp/Helpers/TaskHelper.cs#L28 so we don't break compatibility